sentinel-dashboard/src/main/webapp/resources/app/scripts/directives/header/header.js

```diff
@@ -12,7 +12,20 @@ angular.module('sentinelDashboardApp')
       replace: true,
       controller: function ($scope, $state, $window, AuthService) {
         if (!$window.localStorage.getItem('session_sentinel_admin')) {
-          $state.go('login');
+          AuthService.check().success(function (data) {
+            if (data.code == 0) {
+              $window.localStorage.setItem('session_sentinel_admin', {
+                username: data.data
+              });
+              if (data.data.id == 'FAKE_EMP_ID') {
+                document.getElementById('li-logout').style.display = 'none';
+              } else {
+                document.getElementById('li-logout').style.display = 'block';
```

Comment:
@cdfive Hi , `document.getElementById('li-logout').style.display` This JS code doesn't look very beautiful, can you guide me, how to use angularJS code this.

Reply:
@zhaixiaoxiang We can use `ng-show` directive to show or hide html element.
e.g. `<li ng-show="showLogout">`
The `showLogout` expression can be associated with a variable `$scope.showLogout` in js.
When `showLogout` is true/false, the element is visible/hide.

In header.js:
```javascript
controller: function ($scope, $state, $window, AuthService) {
        $scope.showLogout = true;
        ...
       if (data.data.id == 'FAKE_EMP_ID') {
           $scope.showLogout = true; // show  
       } else {
           $scope.showLogout = true; // hide
       }
}
```

Reply:
There is a small problem:

`auth.enable=false`

When user first visit the page(has no session data in localStorage), the logout button will be hide.
Then refresh page(press F5), the logout button is shown again.

This is due to that the show/hide code is inside the if judgement, `!$window.localStorage.getItem('session_sentinel_admin')`,
when refresh page(press F5), the localStorage has the session data, the if condition is not satisfied so the logout button isn't hide.

It's need to add some code in else branch to handle this case.

Now the code for putting data into localStorage is: 
```javascript
$window.localStorage.setItem('session_sentinel_admin', {
  username: data.data
});
```
When getting data by using `$window.localStorage.getItem('session_sentinel_admin')`,
The json data can't be retrieved, but '[object Object]' instead. 

Maybe we need to optimize it, 
one way is that use `JSON.stringify(xxx))` and `JSON.parse(xxx)`
to put/get data into/from localStorage.

Reply:
Thank you very much for your guidance and code review. I will try this design, but you may be @ wrong person. 😆

Reply:
> Thank you very much for your guidance and code review. I will try this design, but you may be @ wrong person.

OMG, I‘m sorry[facepalm].